client/components/credit-card-form-fields/index.jsx

```diff
@@ -59,54 +80,158 @@ class CreditCardFormFields extends React.Component {
 			[ fieldName ]: maskField( fieldName, previousValue, nextValue ),
 		};
 
-		this.props.onFieldChange( rawDetails, maskedDetails );
+		onFieldChange( rawDetails, maskedDetails );
+	}
+
+	handlePhoneFieldChange = ( { value, countryCode } ) => {
+		this.setState(
+			{
+				phoneCountryCode: countryCode,
+			},
+			() => {
+				this.updateFieldValues( 'phone-number', value );
+			}
+		);
+	};
+
+	handleFieldChange = event => {
+		const { name: fieldName, value: nextValue, options, selectedIndex } = event.target;
+
+		const newState = {};
+
+		if ( fieldName === 'country' ) {
+			newState.countryCode = nextValue;
+			newState.countryName = options[ selectedIndex ].text;
+			if ( ! this.state.phoneCountryCode ) {
+				newState.phoneCountryCode = nextValue;
+			}
+		}
+
+		this.setState( newState, () => this.updateFieldValues( fieldName, nextValue ) );
 	};
 
+	shouldRenderEbanx() {
+		const { countryCode } = this.state;
+		return isEbanx( countryCode );
+	}
+
+	renderEbanxFields() {
+		const { translate, countriesList } = this.props;
+		const { countryCode, phoneCountryCode, countryName } = this.state;
+
+		return [
+			<span key="ebanx-required-fields" className="credit-card-form-fields__info-text">
+				{ translate( 'The following fields are also required for payments in %(countryName)s', {
+					args: {
+						countryName,
+					},
+					context: 'Info text for extra ebanx fields in credit card form',
+				} ) }
+			</span>,
+
+			this.createField( 'document', Input, {
+				label: translate( 'Taxpayer Identification Number', {
+					context:
+						'Individual taxpayer registry identification required ' +
+						'for Brazilian payment methods using EBANX on credit card form',
+				} ),
+				key: 'document',
+			} ),
+
+			this.createField( 'phone-number', FormPhoneMediaInput, {
+				onChange: this.handlePhoneFieldChange,
+				// onBlur: this.handlePhoneFieldChange,
+				countriesList: countriesList,
+				countryCode: phoneCountryCode,
+				label: translate( 'Phone' ),
+				key: 'phone-number',
+			} ),
+
+			this.createField( 'address-1', Input, {
+				maxLength: 40,
+				labelClass: 'credit-card-form-fields__label',
+				label: translate( 'Address' ),
+				key: 'address-1',
+			} ),
+
+			this.createField( 'street-number', Input, {
+				inputMode: 'numeric',
+				label: translate( 'Street Number', {
```

Comment:
:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

Reply:
+1: translator comment is much better here.

Reply:
Thanks. I wasn't aware of the `comment` property.